commons/src/main/java/info/metadude/android/eventfahrplan/commons/SystemClock.kt

```diff
@@ -0,0 +1,112 @@
+package info.metadude.android.eventfahrplan.commons
+
+import org.threeten.bp.*
+import org.threeten.bp.temporal.ChronoField
+
+
+/**
+ * An instance represents a moment in time. All operations are UTC based.
+ * Timezone based dates are retrieved using {#toZonedDateTime}.
+ *
+ * E.g.
+ *
+ * > SystemClock().toZonedDateTime(ZoneOffset.of("GMT+1"))
+ * */
+class SystemClock() : Clock {
+
+    private var time: Instant = Instant.now(org.threeten.bp.Clock.systemUTC())
+    private val utcZone = ZoneOffset.UTC
+
+	private constructor(instant: Instant) : this() {
+		time = instant
+	}
+
+	constructor(milliseconds: Long) : this() {
+        setToMilliseconds(milliseconds)
+    }
+
+	/**
+     * @param UTCDate must be in ISO-8601 format, e.g. "yyyy-MM-dd"
+	 * */
+    constructor(UTCDate: String) : this(LocalDate.parse(UTCDate).atStartOfDay().toInstant(ZoneOffset.UTC))
```

Comment:
# Naming conventions
Please rename `UTCDate` to `utcDate`.

Reply:
Leave as is for now. Maybe rename all places later.

Reply:
Here I would not postpone the renaming since it is a variable which should start with a lower case letter and it is new code.